src/data/list/join.lean

```diff
@@ -24,6 +24,9 @@ attribute [simp] join
 | []       := iff_of_true rfl (forall_mem_nil _)
 | (l :: L) := by simp only [join, append_eq_nil, join_eq_nil, forall_mem_cons]
 
+@[simp] lemma join_singleton {l : list α} : [l].join = l :=
+by simp
```

Comment:
Does that work?
```suggestion
@[simp] lemma join_singleton {l : list α} : [l].join = l := rfl
```

Reply:
No, it doesn't. But `by rw [join, join, append_nil]` does.